### PR TITLE
Implement Issue#899: Include batch update and batch delete into the SqlTransaction 

### PR DESCRIPTION
**Pull request by balanka**
… changing the return typ of the bach delete and update from IO[Exception, List[Int]] to IO[Exception, Int]

### CODE REVIEW

File: jdbc/src/main/scala/zio/sql/SqlDriverLiveModule.scala [hunk 1]
```diff
@@ -41,11 +41,11 @@ trait SqlDriverLiveModule { self: Jdbc =>
         statement.executeUpdate(query)
       }.refineToOrDie[Exception]
 
-    def deleteOnBatch(delete: List[Delete[_]], conn: Connection): IO[Exception, List[Int]] =
+    def deleteOnBatch(delete: List[Delete[_]], conn: Connection): IO[Exception, Int] =
       ZIO.attemptBlocking {
         val statement = conn.createStatement()
         delete.map(delete_ => statement.addBatch(renderDelete(delete_)))
-        statement.executeBatch().toList
+        statement.executeBatch().foldLeft(0) { case (acc, el) => acc + el }
```
Comment (sviezypan) [inline jdbc/src/main/scala/zio/sql/SqlDriverLiveModule.scala:48]: let's simplify foldLeft with .sum

File: jdbc/src/main/scala/zio/sql/SqlDriverLiveModule.scala [hunk 2]
```diff
@@ -58,14 +58,14 @@ trait SqlDriverLiveModule { self: Jdbc =>
...
+    def update(update: List[Update[_]]): IO[Exception, Int] =
       ZIO.scoped(pool.connection.flatMap(updateOnBatch(update, _)))
 
-    def updateOnBatch(update: List[Update[_]], conn: Connection): IO[Exception, List[Int]] =
+    def updateOnBatch(update: List[Update[_]], conn: Connection): IO[Exception, Int] =
       ZIO.attemptBlocking {
         val statement = conn.createStatement()
         update.map(update_ => statement.addBatch(renderUpdate(update_)))
-        statement.executeBatch().toList
+        statement.executeBatch().foldLeft(0) { case (acc, el) => acc + el }
```
Comment (sviezypan) [inline jdbc/src/main/scala/zio/sql/SqlDriverLiveModule.scala:68]: let's simplify foldLeft with .sum

File: postgres/src/test/scala/zio/sql/postgresql/TransactionSpec.scala [hunk 3]
```diff
@@ -38,18 +42,90 @@ object TransactionSpec extends PostgresRunnableSpec with DbSchema {
...
+
+      val result = for {
+        tx <- transact(batchResult)
+      } yield tx
+      assertZIO(result)(equalTo(5)).mapErrorCause(cause => Cause.stackless(cause.untraced))
+    },
+    test("Transaction failed and no row was inserted updated or deleted") {
+      val deleteQuery = deleteFrom(customers).where(verified === false)
+      val id1         = UUID.randomUUID()
+      // val id2 = UUID.randomUUID()
```
Comment (sviezypan) [inline postgres/src/test/scala/zio/sql/postgresql/TransactionSpec.scala:86]: let's clear the comments here and on line 113

**Review by sviezypan — CHANGES_REQUESTED** [review]
hey @balanka thank you for your work. Just few tiny comments.

**Review by jczuchnowski — APPROVED** [review]

**Review by sviezypan — APPROVED** [review]

**Pull request merged**